types/yargs/index.d.ts

```diff
@@ -289,33 +314,82 @@ declare namespace yargs {
     }
 
     interface PositionalOptions {
-        alias?: string | string[];
+        alias?: string | ReadonlyArray<string>;
         choices?: Choices;
         coerce?: (arg: any) => any;
-        conflicts?: string | string[] | { [key: string]: string | string[] };
+        conflicts?: string | ReadonlyArray<string> | { [key: string]: string | ReadonlyArray<string> };
         default?: any;
         desc?: string;
         describe?: string;
         description?: string;
-        implies?: string | string[] | { [key: string]: string | string[] };
+        implies?: string | ReadonlyArray<string> | { [key: string]: string | ReadonlyArray<string> };
         normalize?: boolean;
         type?: PositionalOptionsType;
     }
 
-    interface CommandModule {
-        aliases?: string[] | string;
-        builder?: CommandBuilder;
-        command?: string[] | string;
+    /** Remove keys K in T */
+    type Omit<T, K> = { [key in Exclude<keyof T, K>]: T[key] };
+
+    /** Remove undefined as a possible value for keys K in T */
+    type Defined<T, K extends keyof T> = Omit<T, K> & { [key in K]: Exclude<T[key], undefined> };
+
+    /** Convert T to T[] and T | undefined to T[] | undefined */
+    type ToArray<T> = Array<Exclude<T, undefined>> | Extract<T, undefined>;
+
+    /** Gives string[] if T is an array type, otherwise string. Preserves | undefined. */
+    type ToString<T> = (Exclude<T, undefined> extends any[] ? string[] : string) | Extract<T, undefined>;
+
+    /** Gives number[] if T is an array type, otherwise number. Preserves | undefined. */
+    type ToNumber<T> = (Exclude<T, undefined> extends any[] ? number[] : number) | Extract<T, undefined>;
+
+    type InferredOptionType<O extends Options | PositionalOptions> =
```

Comment:
This is slick.

Reply:
Thanks! And thank you for reviewing this.